### PR TITLE
[Notifications] Fix notifications spamming project secret - part 2

### PR DESCRIPTION
**Pull request by yaelgen**
follow up to: https://jira.iguazeng.com/browse/ML-4498

This PR tackles a space-saving issue with project secrets. Currently, they're all stored in a single secret, which doesn't get cleaned up.
To fix this, we're separating "params" into "params" and "secret_params." This simple change should significantly reduce the space problem.
In the PR linked above, we set the secret key as the hash of the secret_params themselves. That way if the same secret is used multiple times, it won't take up more space than a single instance.

### CODE REVIEW

File: mlrun/__main__.py [hunk 1]
```diff
@@ -1439,7 +1439,7 @@ def add_notification_to_project(
 ):
     for notification_type, notification_params in notification.items():
         project.notifiers.add_notification(
-            notification_type=notification_type, params=notification_params
```
Comment (quaark) [inline mlrun/__main__.py:1442]: not needed here, revert to params :)

**Review by quaark — CHANGES_REQUESTED** [review]
Looks good!
Very minor, also revert everything in CustomNotificationPusher

**Review by liranbg — COMMENTED** [review]
Does it take care of notification secrets from
Older mlrun version (eg 1.4.1)?

**Comment by quaark** [comment]
> Does it take care of notification secrets from
> Older mlrun version (eg 1.4.1)?

yes it does because of how the migrations are done in the db, all the previous params will be secret params

**Review by quaark — APPROVED** [review]
🚀 great stuff

**Pull request merged**